### DISCUSSION TITLE
Integrating Symbolic Regression to get GP Mean Function, in Active Learning.

### DISCUSSION THREAD

**Discussion by utkarshp1161** [ziatdinovmax/gpax#90] [General]
I was playing with https://github.com/ziatdinovmax/gpax/blob/main/examples/GP_sGP.ipynb notebook and I tried one experiment where I fit the mean function using https://github.com/MilesCranmer/PySR and then used that prior in the GP to get the predictive mean and uncertainty[see last section of the notebook: https://file.io/DtTc01chwZ4Q].

I am not sure how useful this will be:
Can we accomodate this in active learning? As we are exploring points we get a new estimation of mean f;n which we use in fitting the gp?

**Comment by ziatdinovmax** [comment]
Hi @utkarshp1161 - it seems to be somewhat similar to [this work](https://github.com/aghosh92/SISSO_sGP), where the mean functions for sGPs in active hypothesis learning came from SISSO analysis (symbolic regression + compressed sensing). Here's the associated [paper](https://pubs.aip.org/aip/aml/article/1/4/046102/2917772/Discovery-of-structure-property-relations-for).

**Reply by utkarshp1161** [reply]
@ziatdinovmax Thanks I will go through it.

**Reply by utkarshp1161** [reply]
I went through briefly https://github.com/aghosh92/SISSO_sGP/blob/main/sGP_AL_QM9_v1b.py . Very interesting work. Please correct me if I am wrong. Here you have defined the set of initial model before active learning starts. In the exploration steps you choose a model using get_best_model_condition(function in code) and then give a reward to the chosen model.

Now, I feel what is different in what I am proposing. I wish to create a new set of models at every exploration step. As at each step we have one extra point from last iteration and thus our choice of models can be a new set, informed by this new data point as well. But the problem I see is how do we carry forward rewards if set of models changing?

**Reply by utkarshp1161** [reply]
Similar example: https://github.com/ziatdinovmax/gpax/blob/main/examples/gpax_hypo.ipynb.

**Comment by ziatdinovmax** [comment]
> Please correct me if I am wrong. Here you have defined the set of initial model before active learning starts.

Yes, that is correct
> I wish to create a new set of models at every exploration step. 

Ah, got it. Yes, that would be a new thing.

**Reply by utkarshp1161** [reply]
Hi max, I am trying to do something similar(but not exactly active learning). Just trying to parse few toy 1d models in automated way. If you can take a look it will be great.

https://drive.google.com/file/d/12GKVWdqQHiFpkhMXdRN8qeNSAWipHnQU/view?usp=sharing

**Comment by utkarshp1161** [comment]
Closing this, as was able to parse the discovered functions as mean function to structured GP. Still trying to think how to go about assigning rewards in active learning.
